Pod/Classes/XExtensionItem.m

```diff
@@ -22,37 +24,50 @@ @implementation XExtensionItemSource
 
 #pragma mark - Initialization
 
-- (instancetype)initWithPlaceholderItem:(id)placeholderItem attachments:(NSArray *)attachments {
+- (instancetype)initWithPlaceholderItem:(id)placeholderItem typeIdentifier:(NSString *)typeIdentifier itemBlock:(XExtensionItemProvidingBlock)activityItemBlock {
     NSParameterAssert(placeholderItem);
     
     self = [super init];
     if (self) {
         _placeholderItem = placeholderItem;
-        _attachments = [attachments copy];
+        _activityItemBlock = [activityItemBlock copy];
         
-        _activityTypesToItemProviderBlocks = [[NSMutableDictionary alloc] init];
-        _activityTypesToSubjects = [[NSMutableDictionary alloc] init];
-        _activityTypesToThumbnailProviderBlocks = [[NSMutableDictionary alloc] init];
+        if (typeIdentifier)
+            _typeIdentifier = [typeIdentifier copy];
+        else
+            _typeIdentifier = [typeIdentifierForActivityItem(placeholderItem) copy];
+        
+        _additionalAttachmentsByActivityType = [[NSMutableDictionary alloc] init];
+        _attributedContentTextByActivityType = [[NSMutableDictionary alloc] init];
     }
     
     return self;
 }
 
-- (instancetype)initWithPlaceholderData:(id)placeholderData
-                     dataTypeIdentifier:(NSString *)dataTypeIdentifier
-                            attachments:(NSArray/*<NSItemProvider>*/ *)attachments {
-    NSParameterAssert(dataTypeIdentifier);
-    
-    self = [self initWithPlaceholderItem:placeholderData attachments:attachments];
-    if (self) {
-        _dataTypeIdentifier = [dataTypeIdentifier copy];
+- (instancetype)initWithURL:(NSURL *)URL {
+    if ([URL isFileURL]) {
```

Comment:
In that declaration `isFileURL` refers to the name of the getter method, which is *the method that is invoked when you access that property*. But the property is called `fileURL`, so that is the only thing that should be used when using property syntax. `URL.isFileURL` will work because the compiler is lenient and lets you use methods as if they are properties, but it is not correct.

See [Apple's documentation](https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html#//apple_ref/doc/uid/TP40011210-CH5-SW2)